pkgs/development/compilers/opensycl/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib, clang15Stdenv, fetchFromGitHub, llvmPackages_15, llvmPackages_rocm, rocm-device-libs, python3, rocm-runtime, cmake, boost, libxml2, libffi, makeWrapper, hip, rocmSupport ? false }:
+let
+  # depending on whether rocm support is enabled or not, we choose a different mkDerivation, postFixup routine and add an extra buildInput
+  mkDerivation = if rocmSupport then llvmPackages_rocm.rocmClangStdenv.mkDerivation else clang15Stdenv.mkDerivation;
+
+  # one could add --hipsycl-targets='omp;hip' here, but I guess this is best left to the user
```

Comment:
Why? Does this work without this?

Reply:
I did make it work without that, but I had to add lld_15 to the runtime path. The rocm version was unable to find lld otherwise.